include/pybind11/pybind11.h

```diff
@@ -1389,6 +1405,69 @@ class module_ : public object {
         //       For Python 2, reinterpret_borrow was correct.
         return reinterpret_borrow<module_>(m);
     }
+
+    /// Must be a POD type, and must hold enough entries for all of the possible slots PLUS ONE for
+    /// the sentinel (0) end slot.
+    using slots_array = std::array<PyModuleDef_Slot, 4>;
```

Comment:
Could we make this `3` in this PR?

And then `4` in the subinterpreter PR.

This would make it clear what's actually needed. E.g. for someone using the subinterpreter diff a couple years down the road, to figure out how to add another option including new slot.

Reply:
Done